src/pkgcheck/checks/python.py

```diff
@@ -193,6 +193,47 @@ def check_depend(self, depend, flags, prefix):
                 return True
         return False
 
+    def check_pep517_depend(self, item):
+        """Check Python ebuilds for missing optional dependencies.
+
+        The problematic case for us is DISTUTILS_OPTIONAL,
+        DISTUTILS_USE_PEP517 != standalone, but no ${DISTUTILS_DEPS} anywhere.
+        """
+        has_distutils_optional = None
+        has_distutils_pep517_non_standalone = None
+
+        for var_node, _ in bash.var_assign_query.captures(item.tree.root_node):
+            var_name = item.node_str(var_node.child_by_field_name('name'))
+
+            if var_name == "DISTUTILS_OPTIONAL":
+                has_distutils_optional = True
+
+            if var_name == "DISTUTILS_USE_PEP517" and not has_distutils_pep517_non_standalone:
+                var_val = item.node_str(var_node.children[-1])
+
+                # For DISTUTILS_USE_PEP517=standalone, the eclass doesn't
+                # provide ${DISTUTILS_DEPS}.
+                has_distutils_pep517_non_standalone = (var_val != "standalone")
+
+            if has_distutils_optional and has_distutils_pep517_non_standalone:
```

Comment:
gpep517 is also required for standalone build systems. I'm thinking of adding another pep517 value in the future to specifically mean "no build system at all" and replace non-pep517 dus=no/manual case.